### PR TITLE
Use alpine as base images.

### LINKED ISSUE #1031: `debian:testing` base image is too unstable for us, as it gets corrupted from time to time
**Describe the bug**
Recently we've migrated to the `debian:testing` base image, however, from yesterday evening it has been corrupted (this time the package repo for the `testing` distro's branch contains corrupted packages, which cannot be installed). This means that this branch is used for testing and we cannot rely on it.

**To Reproduce**
Try to build the PROD image locally without the cache while the Debian repo isn't fixed. Or check the log of the CI job: https://github.com/photoview/photoview/actions/runs/10643539201/job/29507209980?pr=1028

**Expected behavior**
We have to migrate to another base image, stable enough to rely on it. Maybe that will be Alpine, but we need to try it 1st.

 - Are you using Docker? yes 
 - Version `master`

**Additional context**
https://github.com/photoview/photoview/pull/1028

### PR DESCRIPTION
**Pull request by googollee**
Fix #1031 and #1009.

### CODE REVIEW

**Comment by googollee** [comment]
It could provide `amd64,arm64,armv6,armv7` images, see https://github.com/googollee/photoview/pkgs/container/photoview/266580829?tag=build-api.

File: Dockerfile
Comment (kkovaletp) [inline Dockerfile:54]: Why don't we need to source the set env script?

File: Dockerfile
Comment (kkovaletp) [inline Dockerfile:72]: I'd agree that we have to fix this ID by setting it from the unprivileged range, but this will be a breaking change for users, who already created a user and group on their host to match the IDs in the container. Is it unavoidable (like, the 999 is busy)?

File: README.md
Comment (kkovaletp) [inline README.md:257]: There are (were) contributors, who don't want/like to use Docker-based dev env. I guess that we can keep the old instructions for them in the Readme.
Same for lines 300-301

**Review by kkovaletp — COMMENTED** [review]
Some comments and questions:

File: .github/workflows/build.yml
Comment (googollee) [reply .github/workflows/build.yml:38]: Not for testing. It changes variables for both strings and it's easy to change for testing. It's better to keep them.

File: Dockerfile
Comment (googollee) [reply Dockerfile:50]: For `reflex` here, it's safe as it's a pure go binary.

File: Dockerfile
Comment (googollee) [reply Dockerfile:54]: The `set_env` script is used for cross-compiling. As we are building `arm` images in QEMU env, it does not require any cross-compiling. It's better to be removed to keep the workflow simple. Also, it's more stable to rely on the predefined env variables than to set them manually.

File: README.md
Comment (googollee) [reply README.md:257]: It's conflicted with [your reply](https://github.com/photoview/photoview/issues/1031#issuecomment-2323202255). As we are not building on Debian, these packages may not be installed in the future, and we can't verify it easily.

Since we list dependencies above, developers know which packages to be installed in their env.

File: Dockerfile
Comment (googollee) [reply Dockerfile]: OK. Reverted.

File: .github/workflows/build.yml
Comment (googollee) [reply .github/workflows/build.yml]: Updated. Removed `success()` and using `IS_PUSHING_IMAGE` to control both login and push.

File: Dockerfile
Comment (googollee) [reply Dockerfile]: Unfortunately, `alpine` reserves `999`.

```
 > [linux/amd64->arm/v6 release 3/7] RUN chmod +x /app/scripts/install_runtime_dependencies.sh   && addgroup -g 999 photoview   && adduser -u 999 -G photoview -D photoview   && /app/scripts/install_runtime_dependencies.sh:
0.130 addgroup: gid '999' in use
------
------
 > [linux/amd64->arm/v7 release 3/7] RUN chmod +x /app/scripts/install_runtime_dependencies.sh   && addgroup -g 999 photoview   && adduser -u 999 -G photoview -D photoview   && /app/scripts/install_runtime_dependencies.sh:
0.169 addgroup: gid '999' in use
------
------
 > [linux/amd64->arm64 release 3/7] RUN chmod +x /app/scripts/install_runtime_dependencies.sh   && addgroup -g 999 photoview   && adduser -u 999 -G photoview -D photoview   && /app/scripts/install_runtime_dependencies.sh:
0.129 addgroup: gid '999' in use
```

**Comment by googollee** [comment]
@kkovaletp `Imagemagick` in `alpine` doesn't support any camera RAW format by default.

```
/app # magick identify -list format  | grep RAF
/app # magick identify -list format  | grep Fuji
/app # magick identify -list format  | grep Canon
/app # magick identify -list format  | grep NEF
/app # magick identify -list format  | grep CR2
/app # magick identify -list format  | grep Raw
    BAYER* BAYER     rw+   Raw mosaiced samples
   BAYERA* BAYER     rw+   Raw mosaiced and alpha samples
      BGR* BGR       rw+   Raw blue, green, and red samples
     BGRA* BGR       rw+   Raw blue, green, red, and alpha samples
     BGRO* BGR       rw+   Raw blue, green, red, and opacity samples
     CMYK* CMYK      rw+   Raw cyan, magenta, yellow, and black samples
    CMYKA* CMYK      rw+   Raw cyan, magenta, yellow, black, and alpha samples
     GRAY* GRAY      rw+   Raw gray samples
    GRAYA* GRAY      rw+   Raw gray and alpha samples
      M4V  VIDEO     rw+   Raw VIDEO-4 Video
     MONO* MONO      rw-   Raw bi-level bitmap
      RGB* RGB       rw+   Raw red, green, and blue samples
   RGB565* RGB       r--   Raw red, green, blue samples in 565 format
     RGBA* RGB       rw+   Raw red, green, blue, and alpha samples
     RGBO* RGB       rw+   Raw red, green, blue, and opacity samples
    YCBCR* YCBCR     rw+   Raw Y, Cb, and Cr samples
   YCBCRA* YCBCR     rw+   Raw Y, Cb, Cr, and alpha samples
```

I'm working on finding a way to solve it now.

**Comment by googollee** [comment]
Solved [the raw problem](https://github.com/photoview/photoview/pull/1033#issuecomment-2323408915). Also verified that when a cached jpg is missing, rerunning the scan will try to generate a new one.

File: .github/workflows/build.yml
Comment (kkovaletp) [inline .github/workflows/build.yml:37]: the `== 'true'` part is useless, as the variable already contains `true` or `false`

File: Dockerfile
Comment (kkovaletp) [reply Dockerfile:81]: Ok, then this is the time to introduce this breaking change.
This means that the next version would be again major

File: README.md
Comment (kkovaletp) [reply README.md:257]: Ok, I see your point.
Well, for the Docker-based deployments, we provide a fully integrated package (image + PROD compose + DEV compose + optional Makefile). However, on the other side, we still support the direct host deployment scenario, and for that scenario, we have to provide the list of dependencies and deployment steps. Currently, we have the list of supported distros and the migration of our image to Alpine should increase this list, not decrease.
This doesn't mean that we have to provide installation scripts and other easy-install stuff (unless we want to), but we need to provide the list of prerequisites and requirements to make Photoview work on currently supported systems.

In short: in addition to recommending users to use Docker and warn them about the risk of these instructions becoming outdated, please keep the direct installation instructions as well.

> As we are not building on Debian, these packages may not be installed in the future, and we can't verify it easily.

This risk is acceptable for the dev env. You can mention in the warning above, that devs have to be ready to resolve some dependency-related issues and are expected to contribute updates to the readme with the tested corrections.
However, after some time we might see the same type of issues with our PROD host deployment dependency inconsistencies. Since both of us don't use host deployments, the best we can do is to react to users' reports and try to fix those issues. I don't think that this risk should stop us from improving the product.

File: Dockerfile
Comment (kkovaletp) [inline Dockerfile:5]: Does it work for SH shell in the same way? I thought this is Bash-specific modifier)

File: scripts/install_runtime_dependencies.sh
Comment (kkovaletp) [inline scripts/install_runtime_dependencies.sh:6]: As ImageMagick in Alpine repo separates all its libs and plugins, please review the full list of available add-ons and install all image formats and additional libs, which might be useful for us. So far, I found:
- imagemagick-webp
- imagemagick-svg
- imagemagick-jxl
- imagemagick-libs - not sure what are these libs, but maybe they are useful (converters, codecs,...) - please investigate

**Review by kkovaletp — COMMENTED** [review]
Few more comments

File: scripts/install_runtime_dependencies.sh
Comment (kkovaletp) [reply scripts/install_runtime_dependencies.sh:6]: Also, a similar case might be valid for `ffmpeg`: https://pkgs.alpinelinux.org/packages?name=ffmpeg*&branch=edge&repo=&arch=x86_64&maintainer=
However, as we plan to switch to the Jellyfin's ffmpeg, it might be not so critical, but still there might be the case, Jellyfin's ffmpeg is able to use and rely on those add-ons...

File: .github/workflows/build.yml
Comment (googollee) [reply .github/workflows/build.yml:37]: Nope. The ENV value is a `true` string, but not a boolean. `if` requires a boolean. So here we need the comparison.

I tried without the comparison before, and it fails.

File: scripts/install_runtime_dependencies.sh
Comment (googollee) [reply scripts/install_runtime_dependencies.sh]: Done.

I don't think we are able to use `jellyfin-ffmpeg` with `alpine`, because `jellyfin-ffmpeg` binaries are built with `glibc`. `ffmpeg` in `alpine` is compiled with `qsv` and `vaapi` codecs, which are hardware acceleration by Intel and (maybe) AMD CPU. We need to find a way to compile a `ffmpeg` with `cuda` ourselves if we want to support Nvidia.

Comment (kkovaletp) [reply scripts/install_runtime_dependencies.sh]: You've mentioned that you can try to build Jellyfin ffmpeg for Alpine from sources. Isn't that an option anymore?
Did you try to search whether someone already ported Jellyfin ffmpeg to Alpine (or tried at least)?

Yes, with QSV (Intel) and VAAPI (AMD), it is a good start, but it would be much better to have more HW and interfaces/protocols supported

Comment (googollee) [reply scripts/install_runtime_dependencies.sh]: I don't have any options yet. Will research solutions first after this PR.

File: README.md
Comment (googollee) [reply README.md:257]: Another reason of removing those Debian commands is that they are already broken in Debian stable and Ubuntu stable, because the library of go-heif `v1.17` won't be compiled with the `libheif` from the distribution. I can keep those instructions in README, but I guess it just confuses people.

Comment (googollee) [reply README.md]: Added those commands back.

File: .github/workflows/build.yml
Comment (googollee) [reply .github/workflows/build.yml:3]: Reverted.

File: scripts/install_runtime_dependencies.sh
Comment (googollee) [reply scripts/install_runtime_dependencies.sh]: As a quick search, without [Nvidia supporting alpine](https://github.com/NVIDIA/nvidia-container-toolkit/issues/270) , we can't support Nvidia HW.

File: README.md
Comment (kkovaletp) [reply README.md]: > I can keep those instructions in README, but I guess it just confuses people.

Ok, let's see... At least, this will be the last working configuration, so devs could take it as a starting point and fix inconsistencies - that is easier than inventing it from scratch, I guess

File: scripts/install_runtime_dependencies.sh
Comment (kkovaletp) [reply scripts/install_runtime_dependencies.sh]: Sad news(
However, even without NVidia support, Jellyfin still provides a lot more HW support protocols: https://jellyfin.org/docs/general/administration/hardware-acceleration/

I mean, let's merge now with the default ffmpeg + plugins and then it looks to make sense to try replacing it with the Jellyfin's version

**Review by kkovaletp — APPROVED** [review]
Except for the open thread about the breaking change of UID and GID, I don't see any other topics to discuss before the merge, so please let me know what you think on that topic and make sure that everything works as expected (CI builds the image correctly with all the variables applied and product working; the local dev-env is built and running UI and API dev-servers fine with the ability to run tests)

Approving the PR, so once everything is done and checked, it can be merged

File: Dockerfile
Comment (googollee) [reply Dockerfile:81]: Done.

**Pull request merged**
